compiler/src/dmd/importc.d

```diff
@@ -538,6 +538,13 @@ Dsymbol handleSymbolRedeclarations(ref Scope sc, Dsymbol s, Dsymbol s2, ScopeDsy
     if (vd && vd.isCmacro())
         return vd2;
 
+    // function-like C macro? see https://issues.dlang.org/show_bug.cgi?id=24505
+    if (s.isCsymbol() && s.isTemplateDeclaration())
+    {
+        message(s.loc, "function-like macro `%s` conflicts with %s `%s` at %s, not translating to a D template", s.toPrettyChars(), s2.kind(), s2.toPrettyChars(), s2.locToChars());
```

Comment:
Not sure about the use of message here. Either it's valid or not, so silently prefer the function or error.

Reply:
I (non-strongly) wanted to be notified at least, as the macro might be important for proper interop. But if it was wrapped in the .c file as workaround, the message would still appear, so yeah, definitely not ideal.

=> Preferring #16398 instead.

Reply:
> => Preferring #16398 instead.

Oh yeah, wow. I had no idea dmd was doing that (gdc has callbacks for both when the libcpp parser encounters a `define` and `undef` of a macro, then dumps all "live" macro definitions to buffer after reaching end of file - it's a no brainer to do the same).